experimental/packages/opentelemetry-instrumentation-fetch/test/fetch.test.ts

```diff
@@ -239,36 +238,25 @@ describe('fetch', () => {
     );
 
     rootSpan = webTracerWithZone.startSpan('root');
-    api.context.with(api.trace.setSpan(api.context.active(), rootSpan), () => {
+    await api.context.with(api.trace.setSpan(api.context.active(), rootSpan), async () => {
       fakeNow = 0;
-      void getData(fileUrl, method)
-        .then(
-          response => {
-            // this is a bit tricky as the only way to get all request headers from
-            // fetch is to use json()
-            return response.json().then(
-              json => {
-                lastResponse = json;
-                const headers: { [key: string]: string } = {};
-                Object.keys(lastResponse.headers).forEach(key => {
-                  headers[key.toLowerCase()] = lastResponse.headers[key];
-                });
-                lastResponse.headers = headers;
-              },
-              () => {
-                lastResponse = undefined;
-              }
-            );
-          },
-          () => {
-            lastResponse = undefined;
-          }
-        )
-        .then(sinon.clock.runAllAsync)
-        .then(() => {
-          done();
+      try {
+        const responsePromise = getData(fileUrl, method);
+        fakeNow = 300;
+        const response = await responsePromise;
+        // this is a bit tricky as the only way to get all request headers from
+        // fetch is to use json()
+        const json = await response.json();
```

Comment:
That could be assigned straight to `lastResponse`?

Reply:
https://github.com/open-telemetry/opentelemetry-js/pull/2969/commits/6691bac0c5eaaeed2d4d9c6a5c814978ba774873